env0/resource_environment_output_configuration_variable.go

```diff
@@ -0,0 +1,319 @@
+package env0
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type EnvironmentOutputConfigurationVariableParams struct {
+	Name                      string
+	OutputEnvironmentId       string
+	OutputSubEnvironmentAlias string
+	OutputName                string
+	Scope                     string
+	ScopeId                   string
+	Description               string
+	Type                      string
+	IsReadOnly                bool
+	IsRequired                bool
+
+	outputEnvironmentProjectId string
+}
+
+type EnvironmentOutputConfigurationVariableValue struct {
+	OutputName          string `json:"outputName"`
+	ProjectId           string `json:"projectId"`
+	EnvironmentId       string `json:"environmentId,omitempty"`
+	SubEnvironmentAlias string `json:"subEnvironmentAlias,omitempty"`
+}
+
+func resourceEnvironmentOutputConfigurationVariable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceEnvironmentOutputConfigurationVariableCreate,
+		ReadContext:   resourceEnvironmentOutputConfigurationVariableRead,
+		UpdateContext: resourceEnvironmentOutputConfigurationVariableUpdate,
+		DeleteContext: resourceConfigurationVariableDelete,
+		Description:   "for additional details check: https://docs.env0.com/docs/environment-outputs",
+
+		Importer: &schema.ResourceImporter{StateContext: resourceEnvironmentOutputConfigurationVariableImport},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "the name of the variable",
+				Required:    true,
+			},
+			"output_environment_id": {
```

Comment:
cc @yaronya , I'm not sure about that property, and about this resource in general

Reply:
@tomer-landesman - should I pause work on this for now?
(was planning to make necessary changes today).

Reply:
We discussed, the property should remain